File: app/src/main/java/uk/co/mezpahlan/oldtimerag/theguardian/article/ArticleWebViewClient.java
package uk.co.mezpahlan.oldtimerag.theguardian.article;

import android.annotation.TargetApi;
import android.content.Intent;
import android.net.Uri;
import android.os.Build;
import android.webkit.WebResourceRequest;
import android.webkit.WebView;
import android.webkit.WebViewClient;

/**
 * WebViewClient for TheGuardian.Article. Customises the behaviour, look, and feel of the articles.
 */

class ArticleWebViewClient extends WebViewClient {
    private final ArticleFragment articleFragment;

    public ArticleWebViewClient(ArticleFragment articleFragment) {
        this.articleFragment = articleFragment;
    }

    @SuppressWarnings("deprecation")
    @Override
    public boolean shouldOverrideUrlLoading(WebView view, String url) {
        Intent intent = new Intent();
        intent.setAction(Intent.ACTION_VIEW);
        intent.setData(Uri.parse(url));
        if (intent.resolveActivity(articleFragment.getActivity().getPackageManager()) != null) {
            articleFragment.startActivity(intent);
        }
        return true;
    }

    @TargetApi(Build.VERSION_CODES.N)
    @Override
    public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
        Intent intent = new Intent();
        intent.setAction(Intent.ACTION_VIEW);
        intent.setData(Uri.parse(request.getUrl().toString()));
        if (intent.resolveActivity(articleFragment.getActivity().getPackageManager()) != null) {
            articleFragment.startActivity(intent);
        }
        return true;
    }
}

File: app/src/main/java/uk/co/mezpahlan/oldtimerag/theguardian/article/ArticleWebViewClient.kt
package uk.co.mezpahlan.oldtimerag.theguardian.article

import android.annotation.TargetApi
import android.content.Intent
import android.net.Uri
import android.os.Build
import android.webkit.WebResourceRequest
import android.webkit.WebView
import android.webkit.WebViewClient

/**
 * WebViewClient for TheGuardian.Article. Customises the behaviour, look, and feel of the articles.
 */
internal class ArticleWebViewClient(private val articleFragment: ArticleFragment) : WebViewClient() {

    override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
        val intent = Intent()
        intent.action = Intent.ACTION_VIEW
        intent.data = Uri.parse(url)
        if (intent.resolveActivity(articleFragment.requireActivity().packageManager) != null) {
            articleFragment.startActivity(intent)
        }
        return true
    }

    @TargetApi(Build.VERSION_CODES.N)
    override fun shouldOverrideUrlLoading(view: WebView, request: WebResourceRequest): Boolean {
        val intent = Intent()
        intent.action = Intent.ACTION_VIEW
        intent.data = Uri.parse(request.url.toString())
        if (intent.resolveActivity(articleFragment.requireActivity().packageManager) != null) {
            articleFragment.startActivity(intent)
        }
        return true
    }
}
